lib/model/message.dart

```diff
@@ -132,6 +163,55 @@ class MessageStoreImpl extends PerAccountStoreBase with MessageStore {
     }
   }
 
+  @override
+  bool? getEditMessageErrorStatus(int messageId) =>
+    _editMessageRequests[messageId]?.hasError;
+  final Map<int, _EditMessageRequestStatus> _editMessageRequests = {};
+
+  @override
+  void editMessage({
+    required int messageId,
+    required String content,
+  }) async {
+    if (_editMessageRequests.containsKey(messageId)) {
+      throw StateError('message ID already in editMessageRequests');
```

Comment:
nit: `editMessageRequests` isn't quite the name of anything; and the nearest name to that is private so might not be clearest for an error message:

```suggestion
    if (_editMessageRequests.containsKey(messageId)) {
      throw StateError('an edit request is already in progress');
```

(though a private identifier in this message wouldn't be terrible either, since it really should be only developer-facing)